packages/syft/src/syft/core/common/serde/deserialize.py

```diff
@@ -69,18 +70,46 @@ def _deserialize(
         blob.ParseFromString(data_message.content)
 
     # lets try to lookup the type we are deserializing
+    # TODO: This needs to be cleaned up in GenerateWrapper and made more consistent.
+    # There are serveral code paths that come through here and use different ways to
+    # match and overload protobuf -> deserialize type
     obj_type = getattr(type(blob), "schema2type", None)
-    # when a protobuf type is related to multiple classes, it's schema2type will be None.
-    # In that case, we use it's obj_type field.
     if obj_type is None:
+        # TODO: This can probably be removed now we have lists of obj_types
         obj_type = getattr(blob, "obj_type", None)
         if obj_type is None:
             traceback_and_raise(deserialization_error)
         obj_type = index_syft_by_module_name(fully_qualified_name=obj_type)  # type: ignore
         obj_type = getattr(obj_type, "_sy_serializable_wrapper_type", obj_type)
+    elif isinstance(obj_type, list):
```

Comment:
Not the scope of this PR, but we really need to clear the serialize/deserialize functions